pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -169,5 +178,49 @@ public void close()
       throws IOException {
     _indexReader.close();
     _indexDirectory.close();
+    _docIdReaderWriter.close();
+  }
+
+  private class DocIdReaderWriter implements Closeable {
+    private PinotDataBuffer _buffer;
+
+    DocIdReaderWriter(File segmentIndexDir, String column, int numDocs) throws Exception {
+      int length = Integer.BYTES * numDocs;
+      File docIdMappingFile = new File(SegmentDirectoryPaths.findSegmentDirectory(segmentIndexDir),
+          column + LUCENE_TEXT_INDEX_DOCID_MAPPING_FILE_EXTENSION);
+      // For newly added segments, this file will not exist.
+      // For segment refresh, segment reload and server restart, file will exist,
+      // but we don't know if we are here for refresh v/s reload v/s restart.
+      // In case of refresh, we have to build the mapping again, but in case of
+      // reload and restart, we don't. Also, reload has a sub-case where this text index
+      // was indeed created during reload (user enabled on existing or newly added column).
+      // Since there is no way to distinguish why we are here, we build the mapping again
+      // regardless.
+      // TODO: see if we can prefetch the pages
+      _buffer =
+          PinotDataBuffer.mapFile(docIdMappingFile, false, 0, length, ByteOrder.BIG_ENDIAN, getClass().getSimpleName());
```

Comment:
For better performance, I would probably choose native order as this index is always local to one instance?

Reply:
Yes, as discussed using Little Endian and indicated in the comments too. We are unlikely to run on Solaris/Sparc so using LE is fine which is the case on Linux/x86